File: 백준/2748(피보나치 수 2).cpp
/*
Name : Sohyeon Yim
Email : brightcattle@gmail.com 
Blog : https://sohyunwriter.tistory.com/
Problem : https://www.acmicpc.net/problem/2748
*/

#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
	long long n;
	cin >> n;
	long long prev = 0;
	long long cur = 1;
	long long ans;

	if (n == 0)
		ans = 0;
	else if (n == 1)
		ans = 1;
	else {
		while (--n) {
			ans = prev + cur;
			prev = cur;
			cur = ans;
		}
	}
	cout << ans;

	return 0;



}

File: 백준/2748(피보나치 수 2).java
/*
Name : Sohyeon Yim
Email : brightcattle@gmail.com 
Blog : https://sohyunwriter.tistory.com/
Problem : https://www.acmicpc.net/problem/2748
*/

import java.util.Scanner;

public class Main {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int N = sc.nextInt();
		long[] a = new long[N+1];
		a[0] = 0;
		a[1] = 1;
		
		for(int i = 2; i < a.length; i++) {
			a[i] = a[i-1] + a[i-2];
		}
		System.out.println(a[N]);
		
	}

}
